Selenium.HtmlElements/src/Extensions/WebDriverExtensions.cs

```diff
@@ -43,12 +31,14 @@ public static class WebDriverExtensions {
         /// <exception cref="WebDriverTimeoutException">
         ///     Thrown if new tab did not open after 10 seconds
         /// </exception>
-        public static void WaitUntilNewWindowOpened(this IWebDriver webDriver, Action command, String message = null) {
-            webDriver.SerInitialWindowHandls();
+        public static string WaitUntilNewWindowOpened(this IWebDriver webDriver, Action command, String message = null) {
+            var initialWindowHandles = webDriver.WindowHandles;
 
             command();
 
-            webDriver.WaitUntil(s => s.WindowHandles.Count > InitialWindowHandls.Count, message ?? "New browser window did not open after 10 seconds");
+            webDriver.WaitUntil(s => s.WindowHandles.Count > initialWindowHandles.Count, message ?? "New browser window did not open after 10 seconds");
+
+            return webDriver.WindowHandles.Single(windowHandle => !initialWindowHandles.Contains(windowHandle));

```

Comment:
There could be multiple new tabs opened. I'd suggest to return collection of them and let caller do the rest.

Reply:
Fixed.